crates/fuzzing/src/oracles.rs

```diff
@@ -68,24 +58,21 @@ pub fn instantiate(wasm: &[u8], strategy: Strategy) {
 ///
 /// Performs initial validation, and returns early if the Wasm is invalid.
 ///
-/// You can control which compiler is used via passing a `CompilationStrategy`.
-pub fn compile(wasm: &[u8], compilation_strategy: CompilationStrategy) {
+/// You can control which compiler is used via passing a `Strategy`.
+pub fn compile(wasm: &[u8], compilation_strategy: Strategy) {
     if wasmparser::validate(wasm, None).is_err() {
         return;
     }
 
-    let isa = host_isa();
-    let mut compiler = Compiler::new(isa, compilation_strategy);
-    let mut resolver = NullResolver {};
-    let global_exports = Rc::new(RefCell::new(HashMap::new()));
-    let _ = CompiledModule::new(
-        &mut compiler,
-        wasm,
-        None,
-        &mut resolver,
-        global_exports,
-        false,
-    );
+    let engine = Engine::new(&{
+        let mut config = Config::new();
+        config.strategy(compilation_strategy).unwrap();
+        config
+    });
+    let store = Store::new(&engine);
+
+    // This unsafe should be oK since we validated source just before.
```

Comment:
I think the previous validation was just added because it otherwise didn't happen in `CompiledModule`, so could the above validation get removed and the `new_unchecked` changed to `new` here?

Reply:
I think so!

Reply:
Or not? In that case we would ignore compilation errors that happened for valid modules...